SimonImanigirimpuhwe/my-brand-website-api
# #10 ft(article): enable upload  article image

#### What does this PR do?

- Enable uploading article image

#### Description of Task to be completed? 

- import multer to middleware to the put endpoint

- Add articleImage schema property to the article schema

#### How should this be manually tested?

- Clone the repo 

- In your home directory  cd this branch `ft-upload-article-image-#10`

- Then run `npm install` to install all dependencies

- Run `npm run dev` to start the server

- Use postman send a `PUT` request on `http://localhost:3000/articles` 

> In the request body send `{
        title :article title,
       content: article content,
       articleImage
}`

> You should see a returned result with success message and with the updated article


#### Any background context you want to provide?

- To be connected to DB you need to check `env-sample` for the credentials settings

- To be able to update an article you  need to **login**

- To update article image you need to choose in the body  `form-data` and select file

- Multer needs to be installed

#### What are the relevant Trello stories?

- [#10](https://trello.com/c/k7MocvEE/10-upload-article-image)

#### Screenshots (if appropriate)
<img width="1440" alt="article-imag" src="https://user-images.githubusercontent.com/54619678/96647919-d4876b00-132e-11eb-99cf-c172ab758edd.png">


#### Questions:

- N/A
